packages/dds/tree/src/events/events.ts

```diff
@@ -95,7 +93,9 @@ export interface Listenable<TListeners extends object> {
 	 * @param eventName - the name of the event
 	 * @param listener - the handler to run when the event is fired by the emitter
 	 * @returns a {@link Off | function} which will deregister the listener when called.
-	 * This function has undefined behavior if called more than once.
+	 * This deregistration function is idempotent and therefore may be safely called more than once with no effect.
+	 * @remarks Do not register the exact same `listener` object for the same event more than once.
+	 * Doing so will result in undefined behavior.
```

Comment:
I thought we'd decided we wanted to throw in this case (at least for now). If I recall correctly, the idea was that we could always loosen that policy in the future if desired, where it would be harder to add such a policy later.

Reply:
Oh, we did add the error. We should probably note that, rather than saying the behavior is undefined.

Reply:
I see the above thread with Abram. If the idea here is to prevent users from depending on our error behavior, then maybe this makes sense. 

But it raises a higher-level point maybe we should address as a team: do we support consumers using our errors for conditional flow? Is changing an error behavior in a way that is API-non-breaking still a breaking change (e.g. we remove an exceptional case, or change the type of an error returned)? Presumably we don't want to support this, and should document such.

Reply:
My thought was to make the contract as uncommital as possible - "undefined behavior" is even less committal than "we throw". But the actual behavior does throw an error. So, we promise nothing, and also discourage users from using it in practice.

We could also say "it throws an error" as the contract, but that feels slightly less forward-thinking to me. It might seem to imply that we're saying "we like this policy and chose it on purpose as a good design" which we aren't, because we might want to do something less strict later. 

Curious what you think though - it's a subtle difference and maybe not important.

Reply:
Oh, I only saw your first comment when I wrote the above.

I agree with you - if we're not allowed to remove errors as a non-breaking change, I bet we have other problems... I think it should be alright to just say "we error" here and remove it later. I'll update it.